chia/full_node/mempool_check_conditions.py

```diff
@@ -135,6 +135,12 @@ def mempool_check_time_locks(
 
     for spend in bundle_conds.spends:
         unspent = removal_coin_records[bytes32(spend.coin_id)]
+        if spend.birth_height is not None:
+            if spend.birth_height != unspent.confirmed_block_index:
+                return Err.ASSERT_MY_BIRTH_HEIGHT_FAILED
+        if spend.birth_seconds is not None:
+            if spend.birth_seconds != unspent.timestamp:
+                return Err.ASSERT_MY_BIRTH_SECONDS_FAILED
```

Comment:
I'm assuming `ASSERT_MY_BIRTH_SECONDS_FAILED` is from the timestamp in the block, ya?

Reply:
I'm not sure I know what you mean. the `spend` are the conditions imposed on the spend being valid. the `unspent` is the `CoinRecord` of the coin we're spending. The timestamp in the `CoinRecord` is supposed to match the timestamp in the block that confirmed that coin. (in fact, we store this timestamp in the coin_store for every record, which is quite wasteful)

Reply:
Thanks, I meant to say ASSERT_MY_BIRTH_SECONDS. I couldn't find a good place to put this question when I asked it.

Reply:
both `ASSERT_MY_BIRTH_*` conditions ensure that the coin being spent was confirmed on chain at a specific height (or the timestamp of the block at that height)